app/services/annotations/annotate_with_params_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Annotations
+  # This service is mostly needed because some models use protected_attributes and some rely on strong parameters.
+  # Otherwise we could do some magic with the models to transparently `accepts_nested_attributes_for :annotations`.
+  # In this case though I couldn't figure out a clean way, so using a service should make it easier to refactor later.
```

Comment:
At some point we should get rid of protected attributes. They aren't part of rails since long ago and we are using an external gem to keep using them.

Reply:
It doesn't seem like a short term task. Removing service later should be mostly trivial if we remove the protected attributes and figure out a transparent approach.